src/stores/wordpress-versions-slice.ts

```diff
@@ -0,0 +1,104 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import * as Sentry from '@sentry/electron/renderer';
+import { z, ZodError } from 'zod';
+
+const MINIMUM_WORDPRESS_VERSION = '5.9.9';
+
+const wordPressOfferSchema = z.object( {
+	version: z.string(),
+	response: z.enum( [ 'autoupdate', 'upgrade' ] ),
+} );
+
+const wordPressApiResponseSchema = z.object( {
+	offers: z.array( wordPressOfferSchema ),
+} );
+
+interface WordPressVersion {
+	version: string;
+	isBeta: boolean;
+	name: string;
+}
+
+interface WordPressVersionsState {
+	versions: WordPressVersion[];
+	status: 'idle' | 'loading' | 'succeeded' | 'failed';
+	error: string | null;
+}
+
+const initialState: WordPressVersionsState = {
+	versions: [],
+	status: 'idle',
+	error: null,
+};
+
+const extractVersionName = ( version: string ): string => {
+	if ( version.includes( 'beta' ) || version.includes( 'RC' ) ) {
+		return version;
+	}
+	const match = version.match( /^(\d+\.\d+)/ );
+	return match ? match[ 1 ] : version;
+};
+
+export const fetchWordPressVersions = createAsyncThunk(
+	'wordpressVersions/fetchWordPressVersions',
+	async () => {
+		try {
+			const response = await fetch(
+				`https://api.wordpress.org/core/version-check/1.7/?channel=beta&version=${ MINIMUM_WORDPRESS_VERSION }`
+			);
+			if ( ! response.ok ) {
+				throw new Error( 'Failed to fetch WordPress versions' );
+			}
+			const rawData = await response.json();
+			const data = wordPressApiResponseSchema.parse( rawData );
+
+			const offers = data.offers
+				.filter( ( offer ) => offer.response === 'autoupdate' )
+				.map( ( offer ) => ( {
+					version: offer.version,
+					isBeta: offer.version.includes( 'beta' ) || offer.version.includes( 'RC' ),
+					name: extractVersionName( offer.version ),
+				} ) );
+			return offers;
```

Comment:
```suggestion
			return data.offers
				.filter( ( offer ) => offer.response === 'autoupdate' )
				.map( ( offer ) => ( {
					version: offer.version,
					isBeta: offer.version.includes( 'beta' ) || offer.version.includes( 'RC' ),
					name: extractVersionName( offer.version ),
				} ) );
```
Nitpick, but no need for the variable assignment

Reply:
Removed https://github.com/Automattic/studio/pull/986/commits/6a1a4f04c997501d91e57d5c3ffd74f381727dff